### PR TITLE
feat: add ONNX model class support for MLServer

### PR DESCRIPTION
**Pull request by Jooho**
## Summary
- Add `MLServerModelClassONNX` constant (`mlserver_onnx.OnnxModel`) to support ONNX model serving via MLServer
- Add ONNX case to `SetMlServerDefaults()` so that `MLSERVER_MODEL_IMPLEMENTATION` label is correctly set for ONNX models
- Add unit test for ONNX model class in `TestMlServerDefaults`

## Changes
- `pkg/constants/constants.go`: Add `MLServerModelClassONNX = "mlserver_onnx.OnnxModel"` constant
- `pkg/apis/serving/v1beta1/inference_service_defaults.go`: Add ONNX model format case in the switch statement
- `pkg/apis/serving/v1beta1/inference_service_defaults_test.go`: Add ONNX test case to `TestMlServerDefaults`

## Test plan
- [x] Unit test `TestMlServerDefaults` passes with ONNX case
- [ ] Verify no regression for existing model classes (SKLearn, XGBoost, LightGBM, MLFlow)

### CODE REVIEW

**Review by brettmthompson — APPROVED** [review]
/lgtm

**Review by spolti — APPROVED** [review]
/lgtm

**Comment by bartoszmajsak** [comment]
@Jooho is this ODH-specific code?
